### ISSUE TITLE
Facility edit country field

### ISSUE THREAD

**Issue by aethelwulffe** [LibreHealthIO/LibreEHR#981]
Truncates entry to ten characters.

Duplication:
Administration/Facilities.
Add
Enter values, add country name greater than ten characters.
Save, reopen and observe.

It may be that this field expects an abbreviation, and that is desired behavior.  Just want to verify.

**Comment by teryhill** [comment]
If it were a drop down list like in the demographics that would fix the issue, we could store the abbreviation.

**Comment by aethelwulffe** [comment]
Is it worth doing, or is it just a field size thing?  I suppose it matters if it is added to a printed address somewhere.

**Comment by aethelwulffe** [comment]
OK, now I can't edit a facility.  No pop-up or anything...
Official dev-tip demo works.  Not my latest install though.

**Comment by aethelwulffe** [comment]
Could be my test data...this is the freshest dev tip.

**Comment by teryhill** [comment]
It does not look like that code has been touched in a while.

**Comment by aethelwulffe** [comment]
Version from Feb 15 functions fine.
Latest does not.
Same data, same server, different code.
```
An unbalanced tree was written using document.write() causing data from the network to be reparsed. For more information https://developer.mozilla.org/en/Optimizing_Your_Pages_for_Speculative_Parsing  main_info.php:46
This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://developer.mozilla.org/docs/Mozilla/Performance/ScrollLinkedEffects for further details and to join the discussion on related tools and features!  facilities.php
unreachable code after return statement[Learn More]  jquery.1.3.2.js:19:13091
```

**Comment by aethelwulffe** [comment]
Differences:

- new.php
-  login.php
-  main.php
-  immunization_report.php
-  edit_globals.php
-  filters_layout.php
-  patients_tags_layout.php
-  tags_layout.php
-  patient.inc
-   menu_data.json
-  1_0_1-to-2_0_0_upgrade.sql

**Comment by aethelwulffe** [comment]
Never mind...It is the database.  There is something going on when you have a large number of facilities...

**Comment by aethelwulffe** [comment]
UPDATE:
Deleting some facility table rows fixed this issue.  Apparently there is some issue there.  I will try to determine what the limit is, and why.

**Comment by aethelwulffe** [comment]
![image](https://user-images.githubusercontent.com/553751/36497186-8b0c21a2-1708-11e8-9f9e-6b65c9bd6571.png)
Found the problem.  Yes, we have some php warnings that were not there before, but the issue is simply that the pop-up is unviewable with more than 20 or so facilities without zooming.  It becomes uncontrollable after about 30 facilities on a 1080p screen even with CTRL+

**Comment by 2208Abhinav** [comment]
I also observed that when I entered the country as 123456789012345 after saving, the name shortened to 1234567890 (10 characters)

**Comment by 2208Abhinav** [comment]
@aethelwulffe I will try to work on it.

**Comment by 2208Abhinav** [comment]
I am extensively working with the database stuff for my project so that may also help here

**Comment by 2208Abhinav** [comment]
Yes I found it in database we are having **country_code** as **VARCHAR(10)**

**Comment by aethelwulffe** [comment]
Well, 10 characters should be enough for anyone, right?  :)  I the problem must simply be that the field does not record/retrieve the country name via the country code list (in `lists` I think).

**Comment by 2208Abhinav** [comment]
@aethelwulffe you want the country name to be stored in the database with the **country code** name and not the name entered by the user??

**Comment by aethelwulffe** [comment]
If we are using country codes, then that is how we should store/retrieve them.  The real question is "how is this field used".  If it is an important filed for generating/using addresses or phone numbers, then it needs enumerization via the country code list.  If not, then the question remains as to why the field exists at all.  Typically, we have little need for mulitnational clinic support where one clinic has facilities in multiple countries.  The exception is likely to be organizations like the Peace Corps.

**Comment by 2208Abhinav** [comment]
I will find out where we are needing country name for extracting other data

**Comment by aethelwulffe** [comment]
Thanks!  Just as importantly, we should look to WHERE we SHOULD be doing it...

**Comment by teryhill** [comment]
I have also noticed on new installs that the users tend to spell out the state. Georgia for example when they send test claims they get rejected to invalid zip code (don't ask why) but actually the state abbreviation is Ge, if this is changed to Ga everything works. I think the state entry should be changes to the drop down list and the 2 character abbreviation stored.

**Comment by aethelwulffe** [comment]
The georgia thing is obviously an error.

**Comment by 2208Abhinav** [comment]
@teryhill we can provide a drop-down menu and then use an array to covert the option selected to the **country code**

**Comment by teryhill** [comment]
There  is the state options on the misc tab in globals. I don't know it that is even used in the code anywhere any more.

**Comment by 2208Abhinav** [comment]
@teryhill I will pull a request by tomorrow on this issue
